_source/_posts/2022-01-26-webassembly-on-kubernetes-with-rust.md

```diff
@@ -0,0 +1,365 @@
+
+By default, the artifact would be private, but you need to make it public so that you can access it from the Krustlet cluster. Click the package name and click the **Package settings** button, scroll down, click **Change visibility**, and change to **public**.
+
+You can check this by pulling the artifact using the below command:
+
+```bash
+wasm-to-oci pull ghcr.io/<your GitHub user>/rust-wasm:latest
+```
+
+Yay! You have successfully pushed your first Wasm artifact to an OCI registry. Now let's deploy this to the Kubernetes cluster you created earlier.
+
+Create a YAML file, let's say `k8s.yaml`, with the following contents:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: rust-wasi-example
+  labels:
+    app: rust-wasi-example
+  annotations:
+    alpha.wasi.krustlet.dev/allowed-domains: '["https://catfact.ninja/fact"]'
+    alpha.wasi.krustlet.dev/max-concurrent-requests: "42"
+spec:
+  automountServiceAccountToken: false
+  containers:
+    - image: ghcr.io/<your GitHub user>/rust-wasm:latest
+      imagePullPolicy: Always
+      name: rust-wasi-example
+      env:
+        - name: RUST_LOG
+          value: info
+        - name: RUST_BACKTRACE
+          value: "1"
+  tolerations:
+    - key: "node.kubernetes.io/network-unavailable"
+      operator: "Exists"
+      effect: "NoSchedule"
+    - key: "kubernetes.io/arch"
+      operator: "Equal"
+      value: "wasm32-wasi"
+      effect: "NoExecute"
+    - key: "kubernetes.io/arch"
+      operator: "Equal"
+      value: "wasm32-wasi"
+      effect: "NoSchedule"
+```
+
+**Note**: Remember to replace `<your GitHub user>` with your own GitHub username.
+
+The `annotations` and `tolerations` are significant. The annotations are used to allow external network calls from the krustlet, and the tolerations limit the pod to only schedule/run on Wasm nodes. We are also passing some environment variables that the application will use.
+
+Now apply the manifest using the below command and check the pod status.
+
+```bash
+kubectl apply -f k8s.yaml
+
+kubectl get pods -o wide
+# Output
+NAME                READY   STATUS    RESTARTS   AGE     IP       NODE       NOMINATED NODE   READINESS GATES
+rust-wasi-example   1/1     Running   0          6m50s   <none>   krustlet   <none>           <none>
+```
+
+You should see the Wasm workload running successfully on the Krustlet node. Let's check the logs.
+
+```bash
+kubectl logs rust-wasi-example
+# Output
+[2022-01-16T11:42:20Z INFO  rust_wasm] Cat Fact: Polydactyl cats (a cat with 1-2 extra toes on their paws) have this as a result of a genetic mutation. These cats are also referred to as 'Hemingway cats' because writer Ernest Hemingway reportedly owned dozens of them at his home in Key West, Florida.
+[2022-01-16T11:43:21Z INFO  rust_wasm] Cat Fact: The way you treat kittens in the early stages of it's life will render it's personality traits later in life.
```

Comment:
```suggestion
[2022-01-16T11:43:21Z INFO  rust_wasm] Cat Fact: The way you treat a kitten in the early stages of its life will render its personality traits later in life.
```
apostrophes not needed, this is not the contraction "it is" - it's the possessive. Can we help make the cat facts API use better grammar? ;-) maybe not...

Reply:
:rofl:

Reply:
I didn't change its isnce it was output from API, but its fine